### PR TITLE
Clarified and Changed WelcomeInfo Secrets

### PR DESCRIPTION
**Pull request by rozbb**

### CODE REVIEW

**Comment by rozbb** [comment]
The last commit above implements my proposal for using `handshake_secret` & `sender_key_secret` instead of `handshake_key` & `handshake_nonce` for the `WelcomeInfo`s preceding an encrypted `Add`.

I implemented it on top of the wording changes because I think those are important regardless.

Feel free to reject the last commit and just take the first two.

File: draft-ietf-mls-protocol.md [hunk 1]
```diff
@@ -1557,9 +1567,9 @@ struct {
 } RatchetNode;
 
 struct {
-    opaque key<0..255>;
-    opaque nonce<0..255>;
-} KeyAndNonce;
+    opaque handshake_secret<0..255>;
+    opaque sender_key_secret<0..255>;
+} UnframingSecrets;
```
Comment (bifurcation) [inline draft-ietf-mls-protocol.md]: "Unframing" isn't used elsewhere.  Maybe "HandshakeEncryptionSecrets"?

File: draft-ietf-mls-protocol.md [hunk 2]
```diff
@@ -1582,14 +1592,11 @@ In the description of the tree as a list of nodes, the `credential`
...
-Note that the `init_secret` in the Welcome message is the
-`init_secret` at the output of the key schedule diagram in
-{{key-schedule}}.  That is, if the `epoch` value in the Welcome
-message is `n`, then the `init_secret` value is `init_secret_[n]`.
-The new member can combine this init secret with the update secret
-transmitted in the corresponding Add message to get the epoch secret
-for the epoch in which it is added.  No secrets from prior epochs
-are revealed to the new member.
+Note that the `init_secret` in the Welcome message is the `init_secret` of
+the group _after_ the corresponding Add operation has been performed. The
```
[the remark this answers is not in the document]
Comment (rozbb) [reply draft-ietf-mls-protocol.md]: Agreed, I think I worded this very poorly. If I changed _after_ to _before_ would that resolve this?

**Comment by bifurcation** [comment]
Obsoleted by #239

**Pull request closed without merging**